### PR TITLE
Convert to sets

### PR DESCRIPTION
**Pull request by nwlandry**
Converted all instances of edges to sets from lists in response to #158. Changed unit tests to reflect this.

### CODE REVIEW

File: tests/test_convert.py [hunk 1]
```diff
@@ -42,12 +42,12 @@ def test_convert_to_graph(edgelist2, edgelist5):
 
 def test_to_hyperedge_list(edgelist1):
```
Comment (leotrs) [inline tests/test_convert.py:43]: We should change the definitions of `edgelistX` in `conftest.py` to be sets. Then we won't need all those list comprehensions later on.

**Review by leotrs — CHANGES_REQUESTED** [review]
Thanks so much for taking the lead on this one! Does this depend on #161 ?

File: tests/classes/test_hypergraph.py [hunk 2]
```diff
@@ -187,66 +187,66 @@ def test_add_node_to_edge():
...
     assert H.edges.members(dtype=dict) == {
-        "fruits": ["apple", "banana", "pear"],
-        "veggies": ["lettuce"],
+        "fruits": {"apple", "banana", "pear"},
+        "veggies": {"lettuce"},
     }
 
 
 def test_add_edges_from_iterable_of_members():
     edges = [[0, 1], [1, 2], [2, 3, 4]]
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply tests/classes/test_hypergraph.py:196]: Good call.

File: tests/test_convert.py (hunk 1 above)
Comment (nwlandry) [reply tests/test_convert.py:43]: done.

**Review by leotrs — APPROVED** [review]

**Pull request merged**